dbms/src/TiDB/Schema/TiDBSchemaSyncer.h

```diff
@@ -102,6 +102,13 @@ struct TiDBSchemaSyncer : public SchemaSyncer
         return it->second;
     }
 
+    // After all tables have been physical dropped, remove the `cur_version` of given keyspace
+    void removeCurrentVersion(KeyspaceID keyspace_id) override
+    {
+        std::lock_guard lock(schema_mutex);
+        cur_versions.erase(keyspace_id);
+    }
+
     bool syncSchemas(Context & context, KeyspaceID keyspace_id) override
     {
         std::lock_guard lock(schema_mutex);
```

Comment:
Out of topic: Now all keyspace wait on the same `schema_mutex`, which means If a keyspace (ks1) takes a long time to synchronize schema, and another keyspace (ks2) triggers a SchemaSync while processing read requests, the read requests of ks2 may be blocked by ks1.

Maybe we can create an improvement issue for it and resolve it later.

Reply:
New issue https://github.com/pingcap/tiflash/issues/7074